Compfiles/Imo1970P4.lean

```diff
@@ -21,15 +20,781 @@ equals the product of the nubmers in the other set.
+
+      have x_lt_y_or_y_lt_x := Ne.lt_or_lt x_not_y
+
+      have y_in_interval : y ∈ s1 ∪ s2 := by
+        simp
+        right
+        exact y_in_s2
+
+      cases x_lt_y_or_y_lt_x
+      case inl x_lt_y =>
+        have := elems_in_interval_nearby x y n (s1 ∪ s2) partition x_in_interval y_in_interval x_lt_y
+        have nearby : ∃ k ≤ 5, x + k = y := this
+        have p_not_dvd_y := no_other_p_divisors_nearby x y p p_gt_5 x_lt_y nearby p_dvd_x
+        apply p_not_dvd_y
+        exact p_dvd_y
+      case inr y_lt_x =>
+        have := elems_in_interval_nearby y x n (s1 ∪ s2) partition y_in_interval x_in_interval y_lt_x
+        have nearby : ∃ k ≤ 5, y + k = x := this
+        have p_not_dvd_x := no_other_p_divisors_nearby y x p p_gt_5 y_lt_x nearby p_dvd_y
+        apply p_not_dvd_x
+        exact p_dvd_x
+
+    apply p_not_dvd_prod_y
+    exact p_dvd_prod_y
+
+  case inr x_in_s2 =>
+    have := Finset.dvd_prod_of_mem (λ n : ℕ => n) x_in_s2
+    simp only at this
+    have p_dvd_prod_x : p ∣ ∏ m ∈ s2, m := (Dvd.dvd.trans p_dvd_x this)
+
+    have p_dvd_prod_y : p ∣ ∏ m ∈ s1, m := by
+      rw[← equal_products] at p_dvd_prod_x
+      exact p_dvd_prod_x
+
+    have p_not_dvd_prod_y : ¬ p ∣ ∏ m ∈ s1, m := by
+      intro h
+      obtain ⟨y, ⟨y_in_s1, p_dvd_y⟩⟩ := prime_dvd_elem_of p pp s1 h
+
+      have : s2 ∩ s1 = ∅ := by
+        rw[← no_dups]
+        ext x
+        simp
+        simp_all only [Finset.mem_Icc, gt_iff_lt]
+        apply Iff.intro
+        · intro a
+          simp_all only [and_self]
+        · intro a
+          simp_all only [and_self]
+
+      have s2_s1_disjoint : Disjoint s2 s1 := Finset.disjoint_iff_inter_eq_empty.mpr this
+
+      have x_not_y : x ≠ y := (Finset.disjoint_iff_ne.mp s2_s1_disjoint) x x_in_s2 y y_in_s1
+
+      have x_lt_y_or_y_lt_x := Ne.lt_or_lt x_not_y
+
+      have y_in_interval : y ∈ s1 ∪ s2 := by
+        simp
+        left
+        exact y_in_s1
+
+      cases x_lt_y_or_y_lt_x
+      case inl x_lt_y =>
+        have := elems_in_interval_nearby x y n (s1 ∪ s2) partition x_in_interval y_in_interval x_lt_y
+        have nearby : ∃ k ≤ 5, x + k = y := this
+        have p_not_dvd_y := no_other_p_divisors_nearby x y p p_gt_5 x_lt_y nearby p_dvd_x
+        apply p_not_dvd_y
+        exact p_dvd_y
+      case inr y_lt_x =>
+        have := elems_in_interval_nearby y x n (s1 ∪ s2) partition y_in_interval x_in_interval y_lt_x
+        have nearby : ∃ k ≤ 5, y + k = x := this
+        have p_not_dvd_x := no_other_p_divisors_nearby y x p p_gt_5 y_lt_x nearby p_dvd_y
+        apply p_not_dvd_x
+        exact p_dvd_x
+
+    apply p_not_dvd_prod_y
+    exact p_dvd_prod_y
+
+lemma odd_props (n : ℕ) (odd_s : Finset ℕ) (s_odd_eq : odd_s = (Finset.Icc n (n + 5)).filter (λ x => Odd x)) :
+  ∃ (a b c : ℕ), {a, b, c} = odd_s ∧ b = a + 2 ∧ c = b + 2 := by
+  cases Decidable.em (Odd n)
+  case inl h =>
+    have h2 := Odd.not_two_dvd_nat h
+    use n
+    use n + 2
+    use n + 4
+    simp_all
```

Comment:
Please avoid non-terminal simps. Use `simp only` or `simp_all only` instead.
In this particular case,
```
simp only [and_self, and_true]
```
works.

Reply:
Oh I missed the simp_all's, just looked at the simps. Will fix this in the next patch.